app/src/main/java/com/android/sample/model/userAccount/UserAccountViewModel.kt

```diff
@@ -423,6 +423,15 @@ open class UserAccountViewModel(
     }
   }
 
+  fun updateFriendsListAfterRemoval(userId: String) {
+    val updatedList = _friends.value.filterNot { it.userId == userId }
+    _friends.value = updatedList
+  }
+
+  fun updateReceivedListAfterRemoval(userId: String) {
+    val receivedList = _receivedRequests.value.filterNot { it.userId == userId }
+    _receivedRequests.value = receivedList
+  }
```

Comment:
wouldn't it be more efficient to integrate this update directly into the removeFriend() and acceptFriendRequest()/rejectFriendRequest() directly ?

Reply:
Indeed this would be better.